R/data_color.R

```diff
@@ -391,131 +393,118 @@ adjust_luminance <- function(colors,
   # Calculate new luminance values based on a fixed step-change in `x`
   y_2 <- 1 / (1 + exp(-(x + steps)))
 
-  # Rescale new luminance values to [0, 100]
+  # Rescale the new luminance values to [0, 100]
   l <- y_2 * 100.
 
   # Obtain hexadecimal colors from the modified HCL color values
-  hcl_colors <- grDevices::hcl(h, c, l, alpha = NULL)
-
-  # Apply alpha values to the hexadecimal colors
-  hcl_colors <- paste0(hcl_colors, alpha)
-
-  hcl_colors
+  grDevices::hcl(h, c, l, alpha = alpha)
 }
 
-is_hex_col_w_alpha <- function(color) {
-  color %>% tidy_grepl("^#[0-9A-Fa-f]{8}$")
-}
 
-is_hex_col_no_alpha <- function(color) {
-  color %>% tidy_grepl("^#[0-9A-Fa-f]{6}$")
-}
-
-frac_alpha_to_hex <- function(alpha) {
-  grDevices::rgb(0, 0, 0, alpha) %>% substr(8, 9)
-}
-
-#' Extract a vector of alpha values for a vector of colors
+#' Is a string a color value in rgba() format?
+#'
+#' The input for this is a character vector of length 1. The output is a single
+#' logical value.
 #'
 #' @noRd
-get_alpha_vec <- function(colors) {
-
-  vapply(
-    colors,
-    FUN.VALUE = character(1),
-    USE.NAMES = FALSE,
-    function(color) {
-
-      if (color %>% is_hex_col_no_alpha()) {
-        "FF"
-      } else if (color %>% is_hex_col_w_alpha()) {
-        toupper(sub(".*(..)$", "\\1", color))
-      } else if (tolower(color) %in% grDevices::colors()) {
-        "FF"
-      } else {
-        NA_character_
-      }
-    }
-  )
+is_rgba_col <- function(color) {
+  color %>% tidy_grepl("^rgba\\([0-9]+?,[0-9]+?,[0-9]+?,[0-9\\.]+?\\)$")
```

Comment:
There can be whitespace after `(`, before `)`, and before or after `,`.

Reply:
I've changed the regex to: `"^rgba\\(\\s*(?:[0-9]+?\\s*,\\s*){3}[0-9\\.]+?\\s*\\)$"`. Thanks!